src/tests/end-to-end/tests/popup/adhoc-panel.test.ts

```diff
@@ -65,6 +67,34 @@ describe('Ad hoc tools', () => {
         expect(results).toMatchSnapshot();
     });
 
+    it.each(['Automated checks', 'Landmarks', 'Headings', 'Color'])(
+        'should display the pinned target page visualizations when enabling the "%s" toggle',
+        async (toggleAriaLabel: string) => {
+            await gotoAdhocPanel();
+
+            await enableToggleByAriaLabel(toggleAriaLabel);
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
+        },
+    );
+
+    async function enableToggleByAriaLabel(ariaLabel: string): Promise<void> {
+        const toggleSelector = `button[aria-label="${ariaLabel}"]`;
+        const enabledToggleSelector = `${toggleSelector}[aria-checked=true]`;
+        const disabledToggleSelector = `${toggleSelector}[aria-checked=false]`;
+
+        await popupPage.clickSelector(disabledToggleSelector);
+
+        // The toggles will go through a state where they are removed and replaced with a spinner, then re-added to the page
+        // We intentionally omit looking for the loading spinner because it can be fast enough to not be seen by Puppeteer
```

Comment:
Is this the reason for the inconsistency in the `tab-stop` test as well? I added an alternative selector and wait for it to avoid this in my PR.

Reply:
Possibly, but I suspect not; my bet would be that for your issue (and for the flakiness in the high contrast toggle click() working), it's probably going to be that we need to do a better job of making sure the button is actionable before clicking on it. [Cypress has a good overview of the prerequisites it checks for before attempting a click event](https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Actionability); it's probably a good guide for us to follow.